src/decoder/mod.rs

```diff
@@ -923,6 +956,38 @@ impl<R: Read + Seek> Decoder<R> {
         Ok(())
     }
 
+    pub fn read_jpeg(&mut self, read_strip_wise: bool) -> TiffResult<DecodingResult> {
+
+        let offsets: Vec<u32>;
+        let bytes: Vec<u32>;
+        let jpeg_tables: Vec<u8> = self.get_tag_u8_vec(Tag::JPEGTables)?;
+
+        if read_strip_wise {
+            offsets = self.get_tag_u32_vec(Tag::StripOffsets)?;
+            bytes = self.get_tag_u32_vec(Tag::StripByteCounts)?;
+        } else {
+            offsets = self.get_tag_u32_vec(Tag::TileOffsets)?;
+            bytes = self.get_tag_u32_vec(Tag::TileByteCounts)?;
+        }
+
+        let mut res_img = Vec::with_capacity(offsets[0] as usize);
```

Comment:
Allocations should be checked against the memory limits.

Reply:
Also, we haven't validated that `offsets` is not empty. In this case, which is an invalid image, this code would panic but instead should return an error.